app/controllers/team_repositories_controller.rb

```diff
@@ -33,6 +33,21 @@ def destroy
     end
   end
 
+  # POST :team_id/repositories/:repository_id/multiple_update
+  def multiple_update
+    service_call = Repositories::MultipleShareUpdateService.call(repository_id: @repository.id,
+                                                                 user_id: -1,
+                                                                 team_id: current_team.id,
+                                                                 team_ids_for_share: teams_to_share,
```

Comment:
Honestly I'd put those methods in the service and there they can be tested an here you an just pass `multiple_update_params[x]`.